pandas/io/parsers.py

```diff
@@ -1419,6 +1420,56 @@ def __init__(self, kwds):
         # keep references to file handles opened by the parser itself
         self.handles = []
 
+    def _validate_parse_dates_presence(self, columns: List[str]) -> None:
+        """
+        Check if parse_dates are in columns.
+
+        If user has provided names for parse_dates, check if those columns
+        are available.
+
+        Parameters
+        ----------
+        columns : list
+            list of names of the dataframe.
+
+        Raises
+        ------
+        ValueError
+            If column to parse_date is not in dataframe.
+
+        """
+        if isinstance(self.parse_dates, list):
+            # a column in parse_dates could be represented
+            # ColReference = Union[int, str]
+            # DateGroups = List[ColReference]
+            # ParseDates = Union[ DateGroups, List[DateGroups],
+            #     Dict[ColReference, DateGroups]]
+            cols_needed = []
+            for col in self.parse_dates:
+                if isinstance(col, list):
+                    cols_needed.extend(col)
+                else:
+                    cols_needed.append(col)
+        elif isinstance(self.parse_dates, dict):
+            cols_needed = list(chain(*self.parse_dates.values()))
+        else:
+            cols_needed = []
+
+        # get only columns that are references using names (str), not by index
```

Comment:
Is this actually required to do?

Reply:
do you mean the else case?

Reply:
Asking about the isinstance check

Reply:
possible cases, that I thought of and are working fine in current system,

```
>>> parse_dates1 = [["date", "time"]]
>>> parse_dates2 = ["date", "time"]
>>> parse_dates3 = { "date1": ["date", "time"]  }
>>> parse_dates4 = [["date", "time"], "date2" ]
```
In case of dict, we need the values.